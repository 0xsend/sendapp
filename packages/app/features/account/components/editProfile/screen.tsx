import {
  Button,
  Checkbox,
  FadeCard,
  Paragraph,
  ProfileAvatar,
  Separator,
  Spinner,
  SubmitButton,
  XStack,
  YStack,
} from '@my/ui'
import { SchemaForm } from 'app/utils/SchemaForm'
import { ProfileSchema, useProfileMutation } from 'app/utils/useProfileMutation'
import { useUser } from 'app/utils/useUser'
import { UploadAvatar, type UploadAvatarRefObject } from '../uploadProfileImage/screen'
import { useRef, useState } from 'react'
import type { Tables } from '@my/supabase/database.types'
import { Check } from '@tamagui/lucide-icons'
import { SettingsHeader } from 'app/features/account/settings/components/SettingsHeader'
import { ReadOnlyFieldWithLabel } from 'app/features/account/settings/components/ReadOnlyFieldWithLabel'
import { FieldWithLabel } from 'app/features/account/settings/components/FieldWithLabel'
import { useForm } from 'react-hook-form'
import type { z } from 'zod'

enum FormState {
  Overview = 'Overview',
  ProfileForm = 'ProfileForm',
}

export const EditProfile = () => {
  const [formState, setFormState] = useState<FormState>(FormState.Overview)
  const { profile } = useUser()

  return (
    <YStack w={'100%'}>
      <YStack gap={'$3.5'}>
        <SettingsHeader>Profile</SettingsHeader>
        {(() => {
          switch (true) {
            case !profile:
              return <Spinner size="large" />
            case formState === FormState.Overview:
              return (
                <Overview profile={profile} onPress={() => setFormState(FormState.ProfileForm)} />
              )
            case formState === FormState.ProfileForm:
              return (
                <EditProfileForm
                  profile={profile}
                  onSave={() => setFormState(FormState.Overview)}
                />
              )
          }
        })()}
      </YStack>
    </YStack>
  )
}

const Overview = ({ profile, onPress }: { profile: Tables<'profiles'>; onPress: () => void }) => {
  const { name, about, is_public, avatar_url } = profile
  const avatarRef = useRef<UploadAvatarRefObject>(null)

  return (
    <YStack gap={'$5'}>
      <FadeCard>
        <XStack gap={'$5'} width={'100%'}>
          <UploadAvatar ref={avatarRef}>
            <ProfileAvatar
              avatarUrl={avatar_url ? avatar_url : undefined}
              $gtMd={{ size: 88 }}
              size={88}
            />
          </UploadAvatar>
          <YStack gap={'$2'} ai={'flex-start'}>
            <YStack>
              <Paragraph
                size={'$5'}
                fontWeight={500}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Profile Picture
              </Paragraph>
            </YStack>
            <Button unstyled onPress={() => avatarRef.current?.pickImage()}>
              <Button.Text
                textDecorationLine="underline"
                color="$primary"
                $theme-light={{ color: '$color12' }}
                size={'$5'}
              >
                Update
              </Button.Text>
            </Button>
          </YStack>
        </XStack>
        <FieldWithLabel label={'Name'}>
          <Paragraph size={'$8'} fontWeight={'500'}>
            {name || '-'}
          </Paragraph>
        </FieldWithLabel>
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <ReadOnlyFieldWithLabel label={'About'} text={about || '-'} />
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <XStack gap={'$3'} ai={'center'}>
          <Checkbox
            disabled={true}
            checked={!!is_public}
            borderWidth={0}
            backgroundColor={is_public ? '$primary' : '$background'}
            circular={true}
          >
            <Checkbox.Indicator>
              <Check color={'$black'} />
            </Checkbox.Indicator>
          </Checkbox>
          <Paragraph size={'$5'}>{`${is_public ? 'Public' : 'Private'}`} Profile</Paragraph>
        </XStack>
      </FadeCard>
      <SubmitButton theme="green" borderRadius={'$4'} p={'$4'} onPress={onPress}>
        <Button.Text ff={'$mono'} fontWeight={'500'} tt="uppercase" size={'$5'} color={'$black'}>
          edit profile
        </Button.Text>
      </SubmitButton>
    </YStack>
  )
}

function EditProfileForm({ profile, onSave }: { profile: Tables<'profiles'>; onSave: () => void }) {
  const { id, name, about, is_public } = profile
  const { mutateAsync, error } = useProfileMutation(id)
  const form = useForm<z.infer<typeof ProfileSchema>>()

  const handleSubmit = async () => {
    const values = form.getValues()
    await mutateAsync(values)
    onSave()
  }

  return (
    <SchemaForm
      form={form}
      schema={ProfileSchema}
      props={{
        name: {
          'aria-label': 'Name',
          bc: '$color0',
        },
        about: {
          'aria-label': 'Bio',
          placeholder: 'Tell us about yourself',
          backgroundColor: '$color0',
          rows: 2,
          focusStyle: {
            fontStyle: 'normal',
          },
        },
        isPublic: {
          defaultChecked: is_public !== null ? is_public : true,
        },
      }}
      defaultValues={{
        name: name ? name : '',
        about: about ? about : '',
        isPublic: is_public !== null ? is_public : true,
      }}
      formProps={{
        $gtSm: {
          maxWidth: '100%',
        },
      }}
      onSubmit={handleSubmit}
      renderAfter={({ submit }) => (
        <YStack>
          <SubmitButton
            theme="green"
            borderRadius={'$4'}
            p={'$4'}
            mt={'$1'}
            onPress={() => submit()}
          >
            <Button.Text
              ff={'$mono'}
              fontWeight={'500'}
              tt="uppercase"
              size={'$5'}
              color={'$black'}
            >
              SAVE CHANGES
            </Button.Text>
          </SubmitButton>
          {error && (
            <Paragraph marginTop={'$3'} theme="red" color="$color9">
              {error.message}
            </Paragraph>
          )}
        </YStack>
      )}
    >
      {({ name, about, isPublic }) => (
        <FadeCard>
          <FieldWithLabel label={'Name'} gap={'$2'}>
            {name}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <FieldWithLabel label={'About'} gap={'$2'}>
            {about}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <XStack gap={'$3'} ai={'center'}>
            {isPublic}
            <Paragraph size={'$5'}>Make Profile Public</Paragraph>
          </XStack>
        </FadeCard>
      )}
    </SchemaForm>
  )
}
