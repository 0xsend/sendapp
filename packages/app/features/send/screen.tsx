import { useState } from 'react'
import {
  Button,
  Container,
  H1,
  Paragraph,
  Theme,
  XStack,
  YStack,
} from '@my/ui'
import { MainLayout } from 'app/components/layout'
import { IconEthereum } from 'app/components/icons/IconEthereum'
import { Select } from './components/select'
import { NumPad } from './components/numpad'

const items = [
  { name: 'ETH' },
  { name: 'BSC' },
  { name: 'TRON' },
]

export function SendScreen() {
  const [value, setValue] = useState('0.25')

  return (
    <MainLayout>
      <Container>
        <YStack maw={316} pt={'$13'}>
          <XStack jc={'center'}>
            <H1 size={value.length > 4 ? value.length > 8 ? '$9' : '$12' : '$15'}>
              {Number(value).toLocaleString()}
            </H1>
            <XStack pos={'absolute'} r={0} space={'$1.5'}>
              <Paragraph theme={'alt2'}>Bal</Paragraph>
              <Paragraph>1.25</Paragraph>
            </XStack>
          </XStack>
          <XStack jc={'center'} mt={'$6'}>
            <Select items={items} />
          </XStack>
          <NumPad value={value} setValue={setValue} />
          <Button
            my={'$6'}
            py={'$6'}
            borderRadius={'$9'}
            bc={'$backgroundTransparent'}
            boc={'$borderColorFocus'}
          >
            <Paragraph size={'$6'} fontWeight={'700'}>
              Continue
            </Paragraph>
          </Button>
        </YStack>
      </Container>
    </MainLayout>
  )
}
