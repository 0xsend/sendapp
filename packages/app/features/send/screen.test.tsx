import * as solito from 'solito'
import { describe, expect, it } from '@jest/globals'
import { TamaguiProvider, config } from '@my/ui'
import { act, render, screen, userEvent, waitFor } from '@testing-library/react-native'

jest.mock('expo-router', () => ({
  usePathname: jest.fn().mockReturnValue('/send'),
}))

// const params = {}

jest.mock('solito', () => {
  // console.log('mock solito')
  const mockCreateParam = jest.fn(() => {
    // console.log('createParam in')
    return {
      useParam: jest.fn((args) => {
        // console.log('useParam', args)
        return ['test', jest.fn()]
      }),
      useParams: jest.fn((args) => {
        // console.log('useParams', args)
        return ['test', jest.fn()]
      }),
    }
  })
  return {
    __esModule: true,
    createParam: mockCreateParam,
  }
})

jest.mock('app/utils/useProfileLookup')
jest.mock('@my/wagmi')

jest.mock('app/utils/supabase/useSupabase', () => ({
  useSupabase: jest.fn().mockReturnValue({
    rpc: jest.fn().mockReturnValue({
      abortSignal: jest.fn().mockReturnValue({
        data: [
          {
            send_id_matches: [],
            tag_matches: [
              {
                send_id: 3665,
                tag_name: 'test',
                avatar_url: 'https://avatars.githubusercontent.com/u/123',
              },
            ],
            phone_matches: [],
          },
        ],
        error: null,
      }),
    }),
  }),
}))

import { SendScreen } from './screen'

describe('SendScreen', () => {
  it('should render with search', async () => {
    jest.useFakeTimers()

    const tree = render(
      <TamaguiProvider defaultTheme={'dark'} config={config}>
        <SendScreen />
      </TamaguiProvider>
    ).toJSON()

    act(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot('render')

    expect(await screen.findByText('SEARCH BY')).toBeOnTheScreen()
    expect(solito.createParam).toHaveBeenCalled()

    const searchBy = await screen.findByRole('search', { name: 'query' })
    const user = userEvent.setup()

    await act(async () => {
      await user.type(searchBy, 'test')
      jest.advanceTimersByTime(2000)
      jest.runAllTimers()
      await waitFor(() => screen.findByTestId('searchResults'))
    })

    expect(screen.toJSON()).toMatchSnapshot('search')
    expect(screen.getByTestId('tag-search-3665')).toHaveTextContent('??test@test')
    const avatar = screen.getByTestId('avatar')
    expect(avatar).toBeOnTheScreen()
    expect(avatar.props.source.uri).toBe('https://avatars.githubusercontent.com/u/123')
    const link = screen.getByTestId('MockSolitoLink')
    expect(link).toBeOnTheScreen()
    expect(link.props.href).toBe(
      '/send?idType=tag&recipient=test&amount=test&sendToken=test&note=test'
    )
  })
})
