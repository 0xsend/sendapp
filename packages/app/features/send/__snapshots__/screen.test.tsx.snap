// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendScreen should render with search: render 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
      "gap": 32,
      "paddingBottom": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 24,
        "width": "100%",
      }
    }
  >
    <Text
      accessibilityRole="header"
      selectable={true}
      style={
        {
          "color": "hsl(0, 0%, 43.5%)",
          "fontFamily": "System",
          "fontSize": 19.2,
          "fontWeight": "400",
          "lineHeight": 21,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      SEARCH BY
    </Text>
    <View
      style={
        {
          "position": "relative",
        }
      }
      testID="sendSearchContainer"
    >
      <View
        onLayout={[Function]}
        onSubmit={[Function]}
        style={
          {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "gap": 18,
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            animatedStyle={
              {
                "value": {
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                },
              }
            }
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              accessibilityLabel="query"
              accessibilityRole="search"
              control={
                {
                  "_defaultValues": {
                    "query": "",
                  },
                  "_disableForm": [Function],
                  "_executeSchema": [Function],
                  "_fields": {
                    "query": {
                      "_f": {
                        "mount": true,
                        "name": "query",
                        "ref": {
                          "name": "query",
                        },
                        "value": "",
                      },
                    },
                  },
                  "_formState": {
                    "dirtyFields": {},
                    "disabled": false,
                    "errors": {},
                    "isDirty": false,
                    "isLoading": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "submitCount": 0,
                    "touchedFields": {},
                  },
                  "_formValues": {
                    "query": "",
                  },
                  "_getDirty": [Function],
                  "_getFieldArray": [Function],
                  "_getWatch": [Function],
                  "_names": {
                    "array": Set {},
                    "focus": "",
                    "mount": Set {
                      "query",
                    },
                    "unMount": Set {},
                    "watch": Set {
                      "query",
                    },
                    "watchAll": false,
                  },
                  "_options": {
                    "mode": "onSubmit",
                    "reValidateMode": "onChange",
                    "shouldFocusError": true,
                  },
                  "_proxyFormState": {
                    "dirtyFields": false,
                    "disabled": true,
                    "errors": true,
                    "isDirty": false,
                    "isSubmitting": true,
                    "isValid": true,
                    "isValidating": false,
                    "touchedFields": false,
                  },
                  "_removeUnmounted": [Function],
                  "_reset": [Function],
                  "_resetDefaultValues": [Function],
                  "_state": {
                    "action": false,
                    "mount": true,
                    "watch": false,
                  },
                  "_subjects": {
                    "array": {
                      "next": [Function],
                      "observers": [],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "state": {
                      "next": [Function],
                      "observers": [
                        {
                          "next": [Function],
                        },
                        {
                          "next": [Function],
                        },
                      ],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "values": {
                      "next": [Function],
                      "observers": [
                        {
                          "next": [Function],
                        },
                      ],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                  },
                  "_updateDisabledField": [Function],
                  "_updateFieldArray": [Function],
                  "_updateFormState": [Function],
                  "_updateValid": [Function],
                  "getFieldState": [Function],
                  "handleSubmit": [Function],
                  "register": [Function],
                  "setError": [Function],
                  "unregister": [Function],
                }
              }
              focusVisibleStyle={{}}
              focusable={true}
              id=":r0:"
              name="query"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="$Sendtag, Phone, Send ID"
              placeholderTextColor="hsla(191, 32%, 50%, 1)"
              readOnly={false}
              style={
                {
                  "backgroundColor": "hsla(191, 32%, 10%, 1)",
                  "borderBottomColor": "hsla(191, 32%, 32%, 1)",
                  "borderBottomLeftRadius": 9,
                  "borderBottomRightRadius": 9,
                  "borderBottomWidth": 0,
                  "borderLeftColor": "hsla(191, 32%, 32%, 1)",
                  "borderLeftWidth": 0,
                  "borderRightColor": "hsla(191, 32%, 32%, 1)",
                  "borderRightWidth": 0,
                  "borderStyle": "solid",
                  "borderTopColor": "hsla(191, 32%, 32%, 1)",
                  "borderTopLeftRadius": 9,
                  "borderTopRightRadius": 9,
                  "borderTopWidth": 0,
                  "color": "hsl(0, 0%, 93.0%)",
                  "fontFamily": "System",
                  "fontSize": 19.2,
                  "fontStyle": "italic",
                  "fontWeight": "normal",
                  "height": 44,
                  "minWidth": 0,
                  "paddingLeft": 16,
                  "paddingRight": 40,
                  "position": "relative",
                }
              }
              value=""
            />
          </View>
          <View
            style={
              {
                "bottom": 10,
                "gap": 7,
                "position": "absolute",
                "right": 18,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityLabel="Clear input."
        accessibilityRole="button"
        cancelable={true}
        disabled={false}
        focusVisibleStyle={{}}
        focusable={[Function]}
        minPressDuration={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 9,
            "borderBottomWidth": 1,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 1,
            "borderRightColor": "transparent",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 9,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 0,
            "position": "absolute",
            "right": "0",
            "top": "0",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="32"
          bbWidth="32"
          color="#122023"
          focusable={false}
          height={32}
          meetOrSlice={0}
          minX={0}
          minY={0}
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#122023",
                "height": 24,
                "width": 24,
              },
              {
                "flex": 0,
                "height": 32,
                "width": 32,
              },
            ]
          }
          tintColor="#122023"
          vbHeight={32}
          vbWidth={32}
          width={32}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "strokeWidth",
              ]
            }
            strokeWidth="2"
          >
            <RNSVGPath
              d="M8.30006 25.1004L6.90005 23.7004L14.6001 16.0004L6.90005 8.30039L8.30006 6.90039L16.0001 14.6004L23.7001 6.90039L25.1001 8.30039L17.4001 16.0004L25.1001 23.7004L23.7001 25.1004L16.0001 17.4004L8.30006 25.1004Z"
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`SendScreen should render with search: search 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
      "gap": 32,
      "paddingBottom": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 24,
        "width": "100%",
      }
    }
  >
    <Text
      accessibilityRole="header"
      selectable={true}
      style={
        {
          "color": "hsl(0, 0%, 43.5%)",
          "fontFamily": "System",
          "fontSize": 19.2,
          "fontWeight": "400",
          "lineHeight": 21,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      SEARCH BY
    </Text>
    <View
      style={
        {
          "position": "relative",
        }
      }
      testID="sendSearchContainer"
    >
      <View
        onLayout={[Function]}
        onSubmit={[Function]}
        style={
          {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "gap": 18,
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            animatedStyle={
              {
                "value": {
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                },
              }
            }
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              accessibilityLabel="query"
              accessibilityRole="search"
              control={
                {
                  "_defaultValues": {
                    "query": "",
                  },
                  "_disableForm": [Function],
                  "_executeSchema": [Function],
                  "_fields": {
                    "query": {
                      "_f": {
                        "mount": true,
                        "name": "query",
                        "ref": {
                          "name": "query",
                        },
                        "value": "test",
                      },
                    },
                  },
                  "_formState": {
                    "dirtyFields": {
                      "query": true,
                    },
                    "disabled": false,
                    "errors": {},
                    "isDirty": false,
                    "isLoading": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "name": "query",
                    "submitCount": 0,
                    "touchedFields": {
                      "query": true,
                    },
                  },
                  "_formValues": {
                    "query": "test",
                  },
                  "_getDirty": [Function],
                  "_getFieldArray": [Function],
                  "_getWatch": [Function],
                  "_names": {
                    "array": Set {},
                    "focus": "",
                    "mount": Set {
                      "query",
                    },
                    "unMount": Set {},
                    "watch": Set {
                      "query",
                    },
                    "watchAll": false,
                  },
                  "_options": {
                    "mode": "onSubmit",
                    "reValidateMode": "onChange",
                    "shouldFocusError": true,
                  },
                  "_proxyFormState": {
                    "dirtyFields": false,
                    "disabled": true,
                    "errors": true,
                    "isDirty": false,
                    "isSubmitting": true,
                    "isValid": true,
                    "isValidating": false,
                    "touchedFields": false,
                  },
                  "_removeUnmounted": [Function],
                  "_reset": [Function],
                  "_resetDefaultValues": [Function],
                  "_state": {
                    "action": false,
                    "mount": true,
                    "watch": false,
                  },
                  "_subjects": {
                    "array": {
                      "next": [Function],
                      "observers": [],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "state": {
                      "next": [Function],
                      "observers": [
                        {
                          "next": [Function],
                        },
                        {
                          "next": [Function],
                        },
                      ],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "values": {
                      "next": [Function],
                      "observers": [
                        {
                          "next": [Function],
                        },
                      ],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                  },
                  "_updateDisabledField": [Function],
                  "_updateFieldArray": [Function],
                  "_updateFormState": [Function],
                  "_updateValid": [Function],
                  "getFieldState": [Function],
                  "handleSubmit": [Function],
                  "register": [Function],
                  "setError": [Function],
                  "unregister": [Function],
                }
              }
              focusVisibleStyle={{}}
              focusable={true}
              id=":r0:"
              name="query"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="$Sendtag, Phone, Send ID"
              placeholderTextColor="hsla(191, 32%, 50%, 1)"
              readOnly={false}
              style={
                {
                  "backgroundColor": "hsla(191, 32%, 10%, 1)",
                  "borderBottomColor": "hsla(191, 32%, 32%, 1)",
                  "borderBottomLeftRadius": 9,
                  "borderBottomRightRadius": 9,
                  "borderBottomWidth": 0,
                  "borderLeftColor": "hsla(191, 32%, 32%, 1)",
                  "borderLeftWidth": 0,
                  "borderRightColor": "hsla(191, 32%, 32%, 1)",
                  "borderRightWidth": 0,
                  "borderStyle": "solid",
                  "borderTopColor": "hsla(191, 32%, 32%, 1)",
                  "borderTopLeftRadius": 9,
                  "borderTopRightRadius": 9,
                  "borderTopWidth": 0,
                  "color": "hsl(0, 0%, 93.0%)",
                  "fontFamily": "System",
                  "fontSize": 19.2,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "height": 44,
                  "minWidth": 0,
                  "paddingLeft": 16,
                  "paddingRight": 40,
                  "position": "relative",
                }
              }
              value="test"
            />
          </View>
          <View
            style={
              {
                "bottom": 10,
                "gap": 7,
                "position": "absolute",
                "right": 18,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityLabel="Clear input."
        accessibilityRole="button"
        cancelable={true}
        disabled={false}
        focusVisibleStyle={{}}
        focusable={[Function]}
        minPressDuration={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 9,
            "borderBottomWidth": 1,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 1,
            "borderRightColor": "transparent",
            "borderRightWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 9,
            "borderTopWidth": 1,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 44,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 0,
            "position": "absolute",
            "right": "0",
            "top": "0",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="32"
          bbWidth="32"
          color="#122023"
          focusable={false}
          height={32}
          meetOrSlice={0}
          minX={0}
          minY={0}
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#122023",
                "height": 24,
                "width": 24,
              },
              {
                "flex": 0,
                "height": 32,
                "width": 32,
              },
            ]
          }
          tintColor="#122023"
          vbHeight={32}
          vbWidth={32}
          width={32}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "strokeWidth",
              ]
            }
            strokeWidth="2"
          >
            <RNSVGPath
              d="M8.30006 25.1004L6.90005 23.7004L14.6001 16.0004L6.90005 8.30039L8.30006 6.90039L16.0001 14.6004L23.7001 6.90039L25.1001 8.30039L17.4001 16.0004L25.1001 23.7004L23.7001 25.1004L16.0001 17.4004L8.30006 25.1004Z"
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
  <RCTScrollView
    scrollEnabled={true}
    style={
      [
        {
          "gap": 32,
          "marginTop": 40,
        },
        {
          "animatedStyle": {
            "current": {},
          },
          "initial": {
            "updater": [Function],
            "value": {
              "opacity": 0,
              "transform": [
                {
                  "translateY": -10,
                },
              ],
              "width": "100%",
            },
          },
          "viewDescriptors": {
            "add": [Function],
            "remove": [Function],
            "shareableViewDescriptors": {
              "_isReanimatedSharedValue": true,
              "_value": [],
              "addListener": [Function],
              "modify": [Function],
              "removeListener": [Function],
              "value": [],
            },
          },
          "viewsRef": {
            "add": [Function],
            "items": Set {},
            "remove": [Function],
          },
        },
      ]
    }
    testID="searchResults"
  >
    <View>
      <View
        style={
          {
            "flexDirection": "column",
            "gap": 16,
          }
        }
      >
        <Text
          accessibilityRole="header"
          selectable={true}
          style={
            {
              "color": "#B3B3B3",
              "fontFamily": "System",
              "fontSize": 19.2,
              "fontWeight": "400",
              "lineHeight": 21,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "textTransform": "uppercase",
            }
          }
          suppressHighlighting={true}
        >
          tag
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "gap": 24,
            }
          }
        >
          <View
            style={
              {
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "width": "100%",
              }
            }
          >
            <View
              href="/send?idType=tag&recipient=test&amount=test&sendToken=test&note=test"
              testID="MockSolitoLink"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 18,
                  }
                }
                testID="tag-search-3665"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 7,
                      "borderBottomRightRadius": 7,
                      "borderTopLeftRadius": 7,
                      "borderTopRightRadius": 7,
                      "flexDirection": "column",
                      "height": 48,
                      "justifyContent": "center",
                      "maxHeight": 48,
                      "maxWidth": 48,
                      "minHeight": 48,
                      "minWidth": 48,
                      "overflow": "hidden",
                      "position": "relative",
                      "width": 48,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "bottom": 0,
                        "flexDirection": "column",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 1,
                      }
                    }
                  >
                    <Image
                      fullscreen={true}
                      onError={[Function]}
                      onLoad={[Function]}
                      source={
                        {
                          "uri": "https://avatars.githubusercontent.com/u/123",
                        }
                      }
                      style={
                        {
                          "height": 48,
                          "width": 48,
                        }
                      }
                      testID="avatar"
                    />
                  </View>
                  <View
                    style={
                      {
                        "bottom": 0,
                        "flexDirection": "column",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 0,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "borderBottomLeftRadius": 7,
                          "borderBottomRightRadius": 7,
                          "borderTopLeftRadius": 7,
                          "borderTopRightRadius": 7,
                          "flexDirection": "column",
                          "height": 48,
                          "justifyContent": "center",
                          "maxHeight": 48,
                          "maxWidth": 48,
                          "minHeight": 48,
                          "minWidth": 48,
                          "overflow": "hidden",
                          "position": "relative",
                          "width": 48,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "bottom": 0,
                            "flexDirection": "column",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 1,
                          }
                        }
                      >
                        <Image
                          fullscreen={true}
                          onError={[Function]}
                          onLoad={[Function]}
                          source={
                            {
                              "uri": "https://ui-avatars.com/api.jpg?name=test&size=256",
                            }
                          }
                          style={
                            {
                              "height": 48,
                              "width": 48,
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          {
                            "bottom": 0,
                            "flexDirection": "column",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": 0,
                          }
                        }
                      >
                        <Text
                          selectable={true}
                          style={
                            {
                              "color": "hsla(112, 22%, 59%, 1)",
                              "fontFamily": "System",
                              "fontSize": 16.8,
                              "fontWeight": "400",
                              "lineHeight": 19,
                            }
                          }
                          suppressHighlighting={true}
                        >
                          ??
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "gap": 2,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#B3B3B3",
                        "fontFamily": "System",
                        "fontSize": 24,
                        "fontWeight": "300",
                      }
                    }
                    suppressHighlighting={true}
                  >
                    <Text
                      style={
                        {
                          "color": "#FFFFFF",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        }
                      }
                      suppressHighlighting={true}
                    >
                      test
                    </Text>
                  </Text>
                  <Text
                    style={
                      {
                        "color": "hsla(112, 22%, 59%, 1)",
                        "fontFamily": "System",
                        "fontSize": 16.8,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    @test
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
